packages/network/test/integration/message-duplication.test.ts

```diff
@@ -71,27 +66,22 @@ describe('duplicate message detection and avoidance', () => {
             }),
         ])
 
-        const allNodesConnnectedToTrackerPromise = Promise.all(otherNodes.map((node) => {
-            // @ts-expect-error private field
-            return waitForEvent(node.trackerNode, TrackerNodeEvent.CONNECTED_TO_TRACKER)
-        }))
-        // eslint-disable-next-line no-restricted-syntax
-        for (const node of otherNodes) {
-            node.start()
-        }
-        await allNodesConnnectedToTrackerPromise
+        otherNodes.forEach((n) => n.start())
 
-        const allNodesSubscribed = Promise.all(otherNodes.map((node) => {
-            return waitForEvent(node, NodeEvent.NODE_SUBSCRIBED)
-        }))
         // Become subscribers (one-by-one, for well connected graph)
         otherNodes[0].subscribe('stream-id', 0)
         otherNodes[1].subscribe('stream-id', 0)
         otherNodes[2].subscribe('stream-id', 0)
         otherNodes[3].subscribe('stream-id', 0)
         otherNodes[4].subscribe('stream-id', 0)
 
-        await allNodesSubscribed
+        await Promise.all([
```

Comment:
Maybe `Promise.all(otherNodes.map((otherNode) => otherNode.waitForNeighbors('stream-id', 0, 2, 15 * 1000)))`?

Reply:
I think the repetition is low enough here for this to be a rather negligible change.